accepted/0004-undo-redo.md

```diff
@@ -0,0 +1,73 @@
+- Start Date: 2020-01-03
+- RFC PR: #5
+- Mantis Issue: N/A
+
+# Summary
+Create a undo system that is capable of recording the state of OBS actions to
+easily revert the state forwards or backwards in time.
+
+# Motivation
+When working with large and intricate processes - whether it be scenes, sources,
+etc. - accidents are bound to happen. Currently, there is no way to undo
+committed actions. As a result, work can easily be lost at the expense of hours
+of effort.
+
+Implementing a proper undo/redo system would be a drastic increase in
+quality of life for the majority of users.
+
+# Design
+## UX
+The UX would model that of most other applications. Under edit, there would be 2
+options: 'Undo' and 'Redo'. When an action is comitted, for instance deleting a
+scene, the undo button becomes activated, allowing the user to select it.
```

Comment:
I would go one step further, and actually name the undo actions, so that even if the first version doesn't track everything, the user has an idea of what will happen. 

Photoshop example: "Undo New Layer"

This'd require translations for anything that can be undone. Could have a generic "Undo (%1)" translation, with %1 being replaced by the name of the action itself. This'd allow for easy extensibility in the future, and avoid translation duplication of the "Undo" word.